core/java/iesi-core/src/main/java/io/metadew/iesi/script/execution/instruction/data/text/TextSubstring.java

```diff
@@ -47,6 +54,14 @@ public String generateOutput(String parameters) {
             if (start < 0)
                 start += text.length()+1;
 
+
+            //This line must be after checking negative values
+            int countedLines = countNumberOfLines(text,start,inputParameterMatcherTwoArguments.groupCount());
+
+            if(countedLines!=0){
+                start+= countedLines;
+            }
+
```

Comment:
formatting of code. In IntelliJ you can auto format the code. Normally SonarLint should also complain about this formatting. Is the plugin installed at your side?